docs: document automated backport workflow in agent instructions (#57712)
camunda/camunda: AGENTS.md

- Describe why the changes are necessary and note alternatives considered
- Keep descriptions brief and concise
- For bug fixes: ask the engineer whether the fix needs backporting to stable branches before
  opening the PR. If yes, add the appropriate `backport stable/X.Y` label(s) when creating it
  (see "Backporting" below).

### Backporting

Automated by a GitHub Action (`korthout/backport-action`, see `.github/workflows/backport.yml`;
human docs in `CONTRIBUTING.md`). **Never create backport PRs, branches, or cherry-picks yourself,

> and never offer to.** Start a backport only by labelling.

Author: Fixed in 4380550. Softened the trigger line to include the `/backport` comment (not just labels), and corrected `CONTRIBUTING.md` — it now links `korthout/backport-action` (was the stale `zeebe-io` link) and describes conflicts as a committed-markers draft PR rather than a failure, so both docs match the workflow's `draft_commit_conflicts` config.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

- **Start:** add one `backport stable/X.Y` label per target branch (e.g. `backport stable/8.7`).
  Not-yet-merged PR → runs on merge. Already-merged PR → comment `/backport`. Multiple labels →
  multiple backport PRs.
- **Success:** action opens the backport PR; a bot approves and merges it once CI passes.
- **Conflicts:** action opens a **draft** PR with conflict markers committed as-is plus a comment

>   with resolution steps. You may resolve them (check out the branch, fix conflicts and markers) —

Reviewer: From my experience, Claude loves to invent new ways to do this. Stronger language might help:

```suggestion
  with resolution steps. You may resolve them by following these steps exactly —
```
